python_package/madflow/op_classes.py

```diff
@@ -0,0 +1,75 @@
+class function:
+    """function
+    type: void, int, double, ...
+    name: function_name
+    args: function arguments (list of argument objects)
+    scope: function scope (list of strings, one for each line)
+    scope_args: variables defined in the scope (list of argument objects)
+    template: i.e. template <typename T>"""
+
+    def __init__(self, f_type, f_name, f_args, f_scope, f_scope_args, f_template):
+        self.type = f_type
+        self.name = f_name
+        self.args = f_args
+        self.scope = f_scope
+        self.scope_args = f_scope_args
+        self.argn = len(f_args)
+        self.template = f_template
+
+
+class custom_operator:
```

Comment:
Instead, the convention for classes is big camelcase :P 

```suggestion
class CustomOperator:
```

also, you can use for these ones python dataclasses: https://docs.python.org/3/library/dataclasses.html

```python
from dataclasses import dataclass

@dataclass
class CustomOperator:
    """docs"""
    name: str
    scope: str
    functor_name: str
```

And python will generate automatically the `__init__` as well as other goodies (such as nice printing, equality methods, etc).

I would make all functions in this module as dataclasses. When you need to act in some arguments you can add a `__post_init__` method https://docs.python.org/3/library/dataclasses.html#post-init-processing

Reply:
Done.